phpBB/phpbb/members/controller/online.php

```diff
@@ -0,0 +1,618 @@
+
+		// Send data to template
+		$this->template->assign_vars(array(
+			'TOTAL_REGISTERED_USERS_ONLINE'	=> $this->language->lang('REG_USERS_ONLINE', (int) $logged_visible_online, $this->language->lang('HIDDEN_USERS_ONLINE', (int) $logged_hidden_online)),
+			'TOTAL_GUEST_USERS_ONLINE'		=> $this->language->lang('GUEST_USERS_ONLINE', (int) $guest_counter),
+
+			'U_SORT_USERNAME'		=> $this->helper->route('phpbb_members_online', ['sg' => (int) $show_guests, 'sk' => 'a', 'sd' => (($sort_key == 'a' && $sort_dir == 'a') ? 'd' : 'a')]),
+			'U_SORT_UPDATED'		=> $this->helper->route('phpbb_members_online', ['sg' => (int) $show_guests, 'sk' => 'b', 'sd' => (($sort_key == 'b' && $sort_dir == 'a') ? 'd' : 'a')]),
+			'U_SORT_LOCATION'		=> $this->helper->route('phpbb_members_online', ['sg' => (int) $show_guests, 'sk' => 'c', 'sd' => (($sort_key == 'c' && $sort_dir == 'a') ? 'd' : 'a')]),
+
+			'U_SWITCH_GUEST_DISPLAY'	=> $this->helper->route('phpbb_members_online', ['sg' => (int) !$show_guests]),
+			'L_SWITCH_GUEST_DISPLAY'	=> ($show_guests) ? $this->language->lang('HIDE_GUESTS') : $this->language->lang('DISPLAY_GUESTS'),
+			'S_SWITCH_GUEST_DISPLAY'	=> ($this->config['load_online_guests']) ? true : false,
+			'S_VIEWONLINE'				=> true,
+		));
+
+		$this->template->assign_block_vars('navlinks', array(
+			'BREADCRUMB_NAME'	=> $this->language->lang('WHO_IS_ONLINE'),
+			'U_BREADCRUMB'		=> $this->helper->route('phpbb_members_online'),
+		));
+
+		make_jumpbox(append_sid("{$phpbb_root_path}viewforum.$phpEx"));
+
+		// Render
+		return $this->helper->render('viewonline_body.html', $this->language->lang('WHO_IS_ONLINE'));
+	}
+
+	protected function get_location($session_page, $forum_id) // $forum_id can be removed in the future i think https://tracker.phpbb.com/browse/PHPBB3-15434
+	{
+		global $phpbb_container, $phpbb_root_path, $phpEx, $phpbb_adm_relative_path;
+
+		try
+		{
+			$session_page = parse_url($session_page,  PHP_URL_PATH); // Remove query string
+			$session_page = '/' . ((substr($session_page, 0, 8) == 'app.php/') ? substr($session_page, 8) : $session_page); // Remove app.php/
+
+			$match = $this->router->match($session_page);
+
+			switch ($match['_route'])
+			{
+				case 'phpbb_help_bbcode_controller':
+				case 'phpbb_help_bbcode_controller':
+					$location = $this->language->lang('VIEWING_FAQ');
+					$location_url = $this->helper->route('phpbb_help_faq_controller');
+				break;
+
+				case 'phpbb_members_online':
+				case 'phpbb_members_online_whois':
+					$location = $this->language->lang('VIEWING_ONLINE');
+					$location_url = $this->helper->route('phpbb_members_online');
+				break;
+
+				case 'phpbb_report_pm_controller':
+				case 'phpbb_report_post_controller':
+					$location = $this->language->lang('REPORTING_POST');
+					$location_url = append_sid("{$phpbb_root_path}index.$phpEx");
+				break;
+
+				case 'phpbb_message_admin':
+				//case 'phpbb_message_topic':
```

Comment:
Still needed? If not, please remove.

Reply:
Will be, if I don't remember wrong is in another pr